include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -0,0 +1,1071 @@
+
+    return get<pos>(std::move(cfg));
+}
+} // namespace seqan3
+
+namespace std
+{
+
+/*!\brief Value metafunction specialisation for seqan3::detail::configuration
+ * \ingroup core_algorithm
+ * \relates seqan3::detail::configuration
+ * \returns The number of configurations contained in seqan3::detail::configuration.
+ */
+template <typename ... configs_t>
+struct tuple_size<seqan3::detail::configuration<configs_t...>>
+{
+    //!\brief The number of elements.
+    static constexpr size_t value = sizeof...(configs_t);
+};
+
+/*!\brief Value metafunction specialisation for seqan3::detail::configuration.
+ * \ingroup core_algorithm
+ * \relates seqan3::detail::configuration
+ * \returns The the type of configuration for the requested position.
+ */
+template <size_t elem_no, typename ... configs_t>
+struct tuple_element<elem_no, seqan3::detail::configuration<configs_t...>>
+{
+    //!\brief The type of the element at position `elem_no`.
+    using type = meta::at_c<typename seqan3::detail::configuration<configs_t...>::type_list_type, elem_no>;
+};
+
+//!\cond
+//!\brief Provides std::get interface.
+template <size_t elem_no, typename ... configs_t>
+constexpr auto & get(seqan3::detail::configuration<configs_t...> & cfg) noexcept
+{
+    return seqan3::get<elem_no>(cfg);
+}
+
+//!\brief Provides std::get interface.
+template <size_t elem_no, typename ... configs_t>
+constexpr auto const & get(seqan3::detail::configuration<configs_t...> const & cfg) noexcept
+{
+    return  seqan3::get<elem_no>(cfg);
+}
+
+//!\brief Provides std::get interface.
+template <size_t elem_no, typename ... configs_t>
+constexpr auto && get(seqan3::detail::configuration<configs_t...> && cfg) noexcept
+{
+    return seqan3::get<elem_no>(std::move(cfg));
+}
+
+//!\brief Provides std::get interface.
+template <size_t elem_no, typename ... configs_t>
+constexpr auto const && get(seqan3::detail::configuration<configs_t...> const && cfg) noexcept
+{
+    return seqan3::get<elem_no>(std::move(cfg));
+}
+
+//!\brief Provides std::get interface.
+template <typename target_t, typename ... configs_t>
+constexpr auto & get(seqan3::detail::configuration<configs_t...> & cfg) noexcept
+{
+    return seqan3::get<target_t>(cfg);
+}
+
+//!\brief Provides std::get interface.
+template <typename target_t, typename ... configs_t>
+constexpr auto const & get(seqan3::detail::configuration<configs_t...> const & cfg) noexcept
+{
+    return  seqan3::get<target_t>(cfg);
+}
+
+//!\brief Provides std::get interface.
+template <typename target_t, typename ... configs_t>
+constexpr auto && get(seqan3::detail::configuration<configs_t...> && cfg) noexcept
+{
+    return seqan3::get<target_t>(std::move(cfg));
+}
+
+//!\brief Provides std::get interface.
+template <typename target_t, typename ... configs_t>
+constexpr auto const && get(seqan3::detail::configuration<configs_t...> const && cfg) noexcept
+{
+    return seqan3::get<target_t>(std::move(cfg));
+}
```

Comment:
As I said above, I think this code could be simplified dramatically by inheriting from std::tuple or seqan3::pod_tuple or am I missing something?

Reply:
I am not sure it will reduce code dramatically. Actually not sure to expose the positional get Interface at all because of some other issues.  But let’s discuss it tomorrow